docs/install_csi_driver.md

```diff
@@ -0,0 +1,183 @@
+# Nutanix CSI Driver installation with CAPX
+
+The Nutanix CSI driver is fully supported on CAPI/CAPX deployed clusters where all the nodes meet the [Nutanix CSI driver prerequisites](#capi-workload-cluster-prerequisites-for-nutanix-csi-driver).
+
+There are two methods to install the Nutanix CSI driver on a CAPI/CAPX cluster:
+
+- Helm
+- ClusterResourceSet
+
+For more information, check the next sections.
+
+
+
+## CAPI Workload cluster prerequisites for Nutanix CSI Driver
+
+Kubernetes workers need the following prerequisite to use the Nutanix CSI Drivers:
+
+- iSCSI initiator package (for Volumes based block storage)
+- NFS client package ( for Files based storage)
+
+These packages can already be present in the image you use with your Infrastructure provider or you can also rely on your bootstrap provider to install them. More info are available in the [Prerequisites docs](https://portal.nutanix.com/page/documents/details?targetId=CSI-Volume-Driver-v2_5:csi-csi-plugin-prerequisites-r.html).
+
+The packages name and installation method will also vary depending on the OS you plan to use.
+
+In the example below, we will see how to use the `kubeadm` bootstrap provider to deploy these packages on top of an Ubuntu 20.04 image.
+
+The `kubeadm` bootstrap provider allows defining `preKubeadmCommands` that will be launched before kubernetes cluster creation.
+
+These `preKubeadmCommands` can be defined both in `KubeadmControlPlane` for master nodes and in `KubeadmConfigTemplate` for worker nodes.
+
+In our case with an Ubuntu 20.04 image, you need to modify both `KubeadmControlPlane` and `KubeadmConfigTemplate` as in the example below in order to install the prerequisites on all nodes.
+
+```yaml
+spec:
+  template:
+    spec:
+      # .......
+      preKubeadmCommands:
+      - echo "before kubeadm call" > /var/log/prekubeadm.log
+      - apt update
+      - apt install -y nfs-common open-iscsi
+      - systemctl enable --now iscsid
+```
+
+
+
+## Install Nutanix CSI Driver with Helm
+
+The following procedure needs to be applied on a ready workload cluster.
+You can retrieve your workload cluster's kubeconfig and connect with the following command:
+
+```shell
+clusterctl get kubeconfig $CLUSTER_NAME -n $CLUSTER_NAMESPACE > $CLUSTER_NAME-KUBECONFIG
```

Comment:
should we provide sample values instead of env variables like $CLUSTER_NAME-KUBECONFIG etc